source/globalCommands.py

```diff
@@ -4917,6 +4917,33 @@ def script_toggleApplicationsVolumeAdjuster(self, gesture: "inputCore.InputGestu
 	def script_toggleApplicationsMute(self, gesture: "inputCore.InputGesture") -> None:
 		appsVolume._toggleAppsVolumeMute()
 
+	@script(
+		description=_(
+			# Translators: Input help mode message for report language for caret command.
+			"Reports the language for the character where the caret is positioned. "
+			"Pressed twice, shows the character and its language in browse mode",
+		),
+		category=SCRCAT_SYSTEMCARET,
+		speakOnDemand=True,
+	)
+	def script_reeportCaretLanguage(self, gesture):
+		info = self._getTIAtCaret()
+		info.expand(textInfos.UNIT_CHARACTER)
+		curLanguage = self._getCurrentLanguageForTextInfo(info)
+		text = info.text
+		languageDescription = languageHandler.getLanguageDescription(curLanguage)
+		if languageDescription is None:
+			languageDescription = curLanguage
+		repeats = scriptHandler.getLastScriptRepeatCount()
+		if repeats == 0:
+			ui.message(languageDescription)
+		else:
+			# Translators: Character and its language.
+			message = _("{} ({})").format(text, languageDescription)
```

Comment:
I'd only keep the language (as the title of the dialog indicates)

```suggestion
			message = languageDescription
```